src/buffer.c

```diff
@@ -538,7 +538,7 @@ close_buffer(
 	unload_buf = TRUE;
 
 #ifdef FEAT_TERMINAL
-    if (bt_terminal(buf) && (buf->b_nwindows == 1 || del_buf))
+    if (bt_terminal(buf) && (buf->b_nwindows <= 1 || del_buf))
```

Comment:
This is necessary because `close_buffer` gets called multiple times. When using `bunload`, it will be called without `unload_buf` set first, and then called again when `b_nwindows == 0`. This change will make `bunload` work properly.